### PR TITLE
Filter instances that do not have a task

### PR DESCRIPTION
**Pull request by St0rmingBr4in**
To prevent the 'No task in instance' exception

### CODE REVIEW

**Review by kamaradclimber — APPROVED** [review]

File: src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala [hunk 1]
```diff
@@ -73,10 +73,14 @@ private[health] class HealthCheckActor(
...
 
+  def getActiveTaskForInstances(instances: Seq[Instance]): Set[Task.Id] = {
+    instances.filter(_.tasksMap.size != 0).map(_.appTask).filter(_.isActive).map(_.taskId)(collection.breakOut)
+  }
+
   def purgeStatusOfDoneInstances(instances: Seq[Instance]): Unit = {
     logger.debug(s"Purging health status of inactive instances for app ${app.id} version ${app.version} and healthCheck ${healthCheck}")
 
-    val activeTaskIds: Set[Task.Id] = instances.map(_.appTask).filter(_.isActive).map(_.taskId)(collection.breakOut)
+    val activeTaskIds: Set[Task.Id] = getActiveTaskForInstances(instances)
```
Comment (komuta) [inline src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala:83]: There are other calls to `Instance.appTask`, see L117 and L150.

**Review by komuta — CHANGES_REQUESTED** [review]
There are 2 other calls to Instance.appTask we need to protect.

File: src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala (hunk 1 above)
Comment (St0rmingBr4in) [reply src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala:83]: The require in L116 ensure there is one task so L117 should not be a problem there.

Comment (St0rmingBr4in) [reply src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala:83]: Should I add a similar `require` in `checkEnoughInstancesRunning` to protect L150 or just move the L116 to L111 just before the call to `checkEnoughInstancesRunning` ?

Comment (komuta) [reply src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala:83]: Sounds reasonable, let's try this

**Review by komuta — CHANGES_REQUESTED** [review]
The patch is probably good for a try as is, but let's wait until we have dedicated unit tests for merging.
Meanwhile, we can test it by tagging it.

**Review by kamaradclimber — APPROVED** [review]

**Pull request merged**